File: sdkwork-ai-sdk-go/sdk.go
package ai

import (
    "github.com/sdkwork/ai-sdk/api"
    sdkhttp "github.com/sdkwork/ai-sdk/http"
)

type SdkworkAiClient struct {
    http *sdkhttp.Client
    Open *api.OpenApi
    Chat *api.ChatApi
    Video *api.VideoApi
    Thread *api.ThreadApi
    Response *api.ResponseApi
    Rerank *api.RerankApi
    Music *api.MusicApi
    Moderation *api.ModerationApi
    Message *api.MessageApi
    KnowledgeBase *api.KnowledgeBaseApi
    Image *api.ImageApi
    File *api.FileApi
    Embedding *api.EmbeddingApi
    Context *api.ContextApi
    Batch *api.BatchApi
    Audio *api.AudioApi
    Assistant *api.AssistantApi
    Model *api.ModelApi
}

func NewSdkworkAiClient(baseURL string) *SdkworkAiClient {
    cfg := sdkhttp.NewDefaultConfig(baseURL)
    return NewSdkworkAiClientWithConfig(cfg)
}

func NewSdkworkAiClientWithConfig(config sdkhttp.Config) *SdkworkAiClient {
    client := sdkhttp.NewClient(config)
    return &SdkworkAiClient{
        http: client,
        Open: api.NewOpenApi(client),
        Chat: api.NewChatApi(client),
        Video: api.NewVideoApi(client),
        Thread: api.NewThreadApi(client),
        Response: api.NewResponseApi(client),
        Rerank: api.NewRerankApi(client),
        Music: api.NewMusicApi(client),
        Moderation: api.NewModerationApi(client),
        Message: api.NewMessageApi(client),
        KnowledgeBase: api.NewKnowledgeBaseApi(client),
        Image: api.NewImageApi(client),
        File: api.NewFileApi(client),
        Embedding: api.NewEmbeddingApi(client),
        Context: api.NewContextApi(client),
        Batch: api.NewBatchApi(client),
        Audio: api.NewAudioApi(client),
        Assistant: api.NewAssistantApi(client),
        Model: api.NewModelApi(client),
    }
}

func (c *SdkworkAiClient) SetApiKey(apiKey string) *SdkworkAiClient {
    c.http.SetApiKey(apiKey)
    return c
}

func (c *SdkworkAiClient) SetAuthToken(token string) *SdkworkAiClient {
    c.http.SetAuthToken(token)
    return c
}

func (c *SdkworkAiClient) SetAccessToken(token string) *SdkworkAiClient {
    c.http.SetAccessToken(token)
    return c
}

func (c *SdkworkAiClient) SetHeader(key string, value string) *SdkworkAiClient {
    c.http.SetHeader(key, value)
    return c
}

func (c *SdkworkAiClient) Http() *sdkhttp.Client {
    return c.http
}

File: sdkwork-ai-sdk-typescript/src/sdk.ts
import { HttpClient, createHttpClient } from './http/client';
import type { SdkworkAiConfig } from './types/common';
import type { AuthTokenManager } from '@sdkwork/sdk-common';

import { OpenApi, createOpenApi } from './api/open';
import { ChatApi, createChatApi } from './api/chat';
import { VideoApi, createVideoApi } from './api/video';
import { ThreadApi, createThreadApi } from './api/thread';
import { ResponseApi, createResponseApi } from './api/response';
import { RerankApi, createRerankApi } from './api/rerank';
import { MusicApi, createMusicApi } from './api/music';
import { ModerationApi, createModerationApi } from './api/moderation';
import { MessageApi, createMessageApi } from './api/message';
import { KnowledgeBaseApi, createKnowledgeBaseApi } from './api/knowledge-base';
import { ImageApi, createImageApi } from './api/image';
import { FileApi, createFileApi } from './api/file';
import { EmbeddingApi, createEmbeddingApi } from './api/embedding';
import { ContextApi, createContextApi } from './api/context';
import { BatchApi, createBatchApi } from './api/batch';
import { AudioApi, createAudioApi } from './api/audio';
import { AssistantApi, createAssistantApi } from './api/assistant';
import { ModelApi, createModelApi } from './api/model';

export class SdkworkAiClient {
  private httpClient: HttpClient;

  public readonly open: OpenApi;
  public readonly chat: ChatApi;
  public readonly video: VideoApi;
  public readonly thread: ThreadApi;
  public readonly response: ResponseApi;
  public readonly rerank: RerankApi;
  public readonly music: MusicApi;
  public readonly moderation: ModerationApi;
  public readonly message: MessageApi;
  public readonly knowledgeBase: KnowledgeBaseApi;
  public readonly image: ImageApi;
  public readonly file: FileApi;
  public readonly embedding: EmbeddingApi;
  public readonly context: ContextApi;
  public readonly batch: BatchApi;
  public readonly audio: AudioApi;
  public readonly assistant: AssistantApi;
  public readonly model: ModelApi;

  constructor(config: SdkworkAiConfig) {
    this.httpClient = createHttpClient(config);
    this.open = createOpenApi(this.httpClient);

    this.chat = createChatApi(this.httpClient);

    this.video = createVideoApi(this.httpClient);

    this.thread = createThreadApi(this.httpClient);

    this.response = createResponseApi(this.httpClient);

    this.rerank = createRerankApi(this.httpClient);

    this.music = createMusicApi(this.httpClient);

    this.moderation = createModerationApi(this.httpClient);

    this.message = createMessageApi(this.httpClient);

    this.knowledgeBase = createKnowledgeBaseApi(this.httpClient);

    this.image = createImageApi(this.httpClient);

    this.file = createFileApi(this.httpClient);

    this.embedding = createEmbeddingApi(this.httpClient);

    this.context = createContextApi(this.httpClient);

    this.batch = createBatchApi(this.httpClient);

    this.audio = createAudioApi(this.httpClient);

    this.assistant = createAssistantApi(this.httpClient);

    this.model = createModelApi(this.httpClient);
  }

  setApiKey(apiKey: string): this {
    this.httpClient.setApiKey(apiKey);
    return this;
  }

  setAuthToken(token: string): this {
    this.httpClient.setAuthToken(token);
    return this;
  }

  setAccessToken(token: string): this {
    this.httpClient.setAccessToken(token);
    return this;
  }

  setTokenManager(manager: AuthTokenManager): this {
    this.httpClient.setTokenManager(manager);
    return this;
  }

  get http(): HttpClient {
    return this.httpClient;
  }
}

export function createClient(config: SdkworkAiConfig): SdkworkAiClient {
  return new SdkworkAiClient(config);
}

export default SdkworkAiClient;
